### PR TITLE
Semantic segmentation example [in progress]

### PR DESCRIPTION
**Pull request by FreakTheMighty**
I've combined the the two branches you created so that I can start to build `fcn8_vgg.py` with tflearn. tflearn already has a vgg example, so that's my starting place.

I have a few questions that you might be able to answer for me.

What role do the [pred operations](https://github.com/MarvinTeichmann/tensorflow-fcn/blob/master/fcn8_vgg.py#L108) play?

The loss function you wrote, do you think it belongs with tflearn's loss functions, or is it what tflearn calls an [objective](http://tflearn.org/objectives/)?

### CODE REVIEW

**Comment by MarvinTeichmann** [comment]
> What role do the pred operations play?

The pred operations are used in inference only. They are not needed for training. The pred operations return the class predicted for each pixel and are used in [test_fcn32_vgg.py](https://github.com/MarvinTeichmann/tensorflow-fcn/blob/master/test_fcn32_vgg.py) for visualization purpose.

> The loss function you wrote, do you think it belongs with tflearn's loss functions, or is it what tflearn calls an objective?

Yes, it is.

> I've combined the the two branches you created so that I can start to build fcn8_vgg.py with tflearn. 

Sounds reasonable. Your solution, applying a merge commit on top of the feature branch `semantic_segmentation_loss` is not optimal. To keep the git history nice and clean I created a new feature branch `fcn_layers` and [cherry-picked](https://git-scm.com/docs/git-cherry-pick) 192bb3a42fb029d7f61aa8a4051e8897453ad8e3 as well as 138e58221697afdf4a392cfeb8facc2e8c64a5af in that branch. 

Also I think 3b2837388f2680a564b30a84452f6cc667db4dec should not be merged into `semantic_segmentation_loss`. This is a different feature. I would suggest, that you apply 3b2837388f2680a564b30a84452f6cc667db4dec on top of `fcn_layers` in a new feature branch.

Be aware that I have amend and forced-pushed 138e582. (Moved `weak_cross_entropy_2d` to `objective.py`). So be careful to not fetch and old version of this commit!

> tflearn already has a vgg example, so that's my starting place.

Sounds good! Good luck with that.

**Comment by MarvinTeichmann** [comment]
This branch is now merged into master and no longer active. Please apply your commits on top of master and start a new pull request. The feature branch  `semantic_segmentation_loss` will be deleted soon.
